### PR TITLE
[GREB-134] Add coupon redeem by max redemption fields

### PR DESCRIPTION
**Pull request by MayaLi**
# [GREB-134] 

LocalStripe needs to expose `max_redemption` and `redeem_by` fields 

[GREB-134]: https://calmdotcom.atlassian.net/browse/GREB-134?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: localstripe/resources.py [hunk 1]
```diff
@@ -542,32 +542,54 @@ class Coupon(StripeObject):
...
     def __init__(self, id=None, name=None, duration=None, amount_off=None,
                  percent_off=None, currency=None, metadata=None,
-                 duration_in_months=None, **kwargs):
+                 duration_in_months=None, max_redemptions=None, redeem_by=None, **kwargs):
         if kwargs:
             raise UserError(400, 'Unexpected ' + ', '.join(kwargs.keys()))
 
         amount_off = try_convert_to_int(amount_off)
-        percent_off = try_convert_to_int(percent_off)
+        percent_off = try_convert_to_float(percent_off)
```
Comment (MayaLi) [inline localstripe/resources.py:551]: https://stripe.com/docs/api/coupons/object#coupon_object-percent_off documentation says percent_off is decimal

File: localstripe/resources.py [hunk 2]
```diff
@@ -542,31 +542,52 @@ class Coupon(StripeObject):
 
     def __init__(self, id=None, name=None, duration=None, amount_off=None,
                  percent_off=None, currency=None, metadata=None,
-                 duration_in_months=None, **kwargs):
+                 duration_in_months=None, max_redemptions=None, redeem_by=None, **kwargs):
```
Comment (vincentcr) [inline localstripe/resources.py]: nit: can you align this to 78 chars pls.

File: localstripe/resources.py [hunk 3]
```diff
@@ -542,31 +542,52 @@ class Coupon(StripeObject):
...
-                assert type(percent_off) is float or type(percent_off) is int
-                assert percent_off >= 0 and percent_off <= 100
-            assert duration in ('forever', 'once', 'repeating')
+                assert type(percent_off) is float or type(percent_off) is int, \
+                    f'percent_off expected to be of type float or int, got: {percent_off}'
+                assert 0 <= percent_off <= 100, \
+                    f'percent_off must be between 0 and 100, got: {percent_off}'
             if amount_off is not None:
-                assert type(currency) is str and currency
+                assert type(currency) is str and currency, f'currency is expected, got: {currency}'
```
Comment (vincentcr) [inline localstripe/resources.py]: nit: align to 78 chars (put error string on second line with `\`)

**Review by vincentcr — APPROVED** [review]
Looks good overall. Only a few formatting nits.

**Review by AriGonzoCalm — APPROVED** [review]

**Pull request merged**